nextcloud/server
# fix(userstatus): set user status to 'In a meeting' if calendar is busy

* Fixes: https://github.com/nextcloud/server/issues/31177

## Summary

A calendar query looks at the calendar to see if there are any events. TRANSPARENT calendars are ignored.

We ignore events if they're
- an OOO calendar event as that status is already handled
- a TRANSPARENT event
- an event that started before the user has set a custom status

We also do not overwrite a "DND" or "INVISIBLE" status or the Talk "In a Call" message.

## How to test

- For OOO: set an OOO period and see that the OOO status stays persistent and is not overwritten by the calendar automation
- For calendar events: set an event in your calendar. Wait until it begins and wait until after it ends, status should be set to "In a meeting" and reset after. I use 5 / 10 minute events to test.
- For transparent calendar events - the same, see that the staus DOES NOT change to "In a meeting"
- For all day events, the status should stay "In a meeting" at all times unless overwritten by a custom status and should not revert back
- Overwriting with a custom status: See "In a meeting" status, set a custom status, reload. See that custom status still exists until next event.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
